src/artwork.c

```diff
@@ -440,14 +499,29 @@ artwork_read_byurl(struct evbuffer *evbuf, const char *url)
     }
 
   content_type = keyval_get(kv, "Content-Type");
-  if (content_type && (strcmp(content_type, "image/jpeg") == 0))
+
+  if (content_type && (strcmp(content_type, "application/json") == 0))
+    ret = artwork_read_json_byurl(evbuf, url);
+  else if ( content_type && ( strcmp(content_type, "image/jpeg") == 0 || strcmp(content_type, "image/jpg") == 0 ) )
     ret = ART_FMT_JPEG;
   else if (content_type && (strcmp(content_type, "image/png") == 0))
     ret = ART_FMT_PNG;
   else
-    {
-      DPRINTF(E_LOG, L_ART, "Artwork from '%s' has no known content type\n", url);
-      goto error;
+    { // With no content-type see if we can guess it from url (does it have .jpg or .png)
```

Comment:
I don't think we should do this. Server must set content-type.

Reply:
Agree, will remove.  It's how the existing code was executing, I just moved it all checking logic to one place.